app/client/cli/debug.go

```diff
@@ -227,3 +233,27 @@ func sendDebugMessage(debugMsg *messaging.DebugMessage) {
 		logger.Global.Fatal().Err(err).Msg("Failed to send debug message")
 	}
 }
+
+func getP2PModule(
+	runtimeMgr *runtime.Manager,
+	addrBookProvider addrbook_provider.AddrBookProvider,
+	currentHeightProvider current_height_provider.CurrentHeightProvider,
+) (modules.P2PModule, error) {
+	p2pConfig := runtimeMgr.GetConfig().P2P
+	bus := runtimeMgr.GetBus()
+
+	var (
+		mod modules.Module
+		err error
+	)
+	if p2pConfig.UseLibP2P {
+		mod, err = libp2p.CreateWithProviders(bus, addrBookProvider, currentHeightProvider)
+	} else {
+		mod, err = p2p.CreateWithProviders(bus, addrBookProvider, currentHeightProvider)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	return mod.(modules.P2PModule), nil
+}
```

Comment:
This doesn't quite work as written because both p2p modules' `CreateWithProviders` functions returns a `modules.Module` interface which can't be assigned to the return variable directly because the return type is `modules.P2PModule`. I also do have a strong opinion against "naked returns" (as I've heard them called) as I was taught to avoid this when I was learning go. In my opinion, naked returns detract from readability and can become troublesome when copy/pasting or moving code such that it can negatively impact developer experience.

Reply:
I'll take it